### PR TITLE
Set labels on templated discussions

### PR DESCRIPTION
**Pull request by taylorcuilty**
Closes https://github.com/github/discussions/issues/2653

### CODE REVIEW

File: .github/workflows/label-templated-discussions.yml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+                    node {
+                      id
+                      name
+                    }
+                  }
+                }
+              }
+            }' > repository_label_data.json
+
+          echo 'LABEL_ID='$(jq --arg topic $TOPIC '.data.repository.labels.edges[].node | select(.name== $topic) | .id' repository_label_data.json) >> $GITHUB_ENV
```
Comment (taylorcuilty) [inline .github/workflows/label-templated-discussions.yml:73]: When I was testing locally, `select(.name== $TOPIC)` didn't work, I had to pass it in as an arg like this

File: .github/workflows/label-templated-discussions.yml [hunk 2]
```diff
@@ -73,7 +73,7 @@ jobs:
           echo 'LABEL_ID='$(jq '.data.repository.labels.edges[].node | .id ' repository_label_data.json) >> $GITHUB_ENV
 
       - name: Label Discussion
-        if: ${{ steps.get_selected_topic.outputs.result != '' }}
+        if: ${{ steps.get_selected_topic.outputs.result != '' }} && $LABEL_ID
```
Comment (taylorcuilty) [inline .github/workflows/label-templated-discussions.yml:76]: Not sure about this syntax

**Review by Craw69699 — COMMENTED** [review]
How do I do this

**Comment by Craw69699** [comment]
@Craw69699

File: .github/workflows/label-templated-discussions.yml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+jobs:
+  label-templated-discussion:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Get selected topic
+        id: get_selected_topic
+        uses: actions/github-script@v6
+        env:
+          DISCUSSION_HTML: ${{ github.event.discussion.html }}
```
Comment (taylorcuilty) [inline .github/workflows/label-templated-discussions.yml:16]: Needs an additional step to fetch the discussion body - it's not included in the payload.

File: .github/workflows/label-templated-discussions.yml [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        with:
+          script: |
+            try {
+              const headings = $DISCUSSION_HTML.getElementsByTagName("h3");
```
Comment (gnfisher) [inline .github/workflows/label-templated-discussions.yml:20]: Would it be helpful for the the category form template to set an expected `id` on the element we are looking for, so we can just look for that `id` and return if its not available?

Comment (taylorcuilty) [reply .github/workflows/label-templated-discussions.yml:20]: @gnfisher yes that would be ideal

Comment (taylorcuilty) [reply .github/workflows/label-templated-discussions.yml]: Unfortunately the ids in the template don't end up in the discussion body html, so we had to go back to looking for the topic based on it's position https://github.com/community/community/pull/38309/commits/c295521cd0d6f9285e1ed8fc529f314a255864ec

**Comment by ks0304725** [comment]
Gi

**Comment by Bestra** [comment]
Is there a demo of this we can see? If we merge it and things go awry, what's the worst that'll happen? The action will just fail?

**Comment by taylorcuilty** [comment]
> Is there a demo of this we can see?
I've been testing this in a separate private test repo. I'll record a demo and add it here! Thanks for the suggestion ❤️ 

> If we merge it and things go awry, what's the worst that'll happen? The action will just fail?

If the action fails, the discussion will still get created, it just won't have the additional label

**Review by Bestra — APPROVED** [review]
This worked great for me in a test repo 🚢  🎉 

https://user-images.githubusercontent.com/2043348/200678969-d9f07b77-534d-4f9e-905e-386af7e410f3.mp4

**Pull request merged**